blackboxopt/optimization_loops/file_based_distributed.py

```diff
@@ -0,0 +1,160 @@
+# Copyright (c) 2024 - for information on the respective copyright owner
+# see the NOTICE file and/or the repository https://github.com/boschresearch/blackboxopt
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import glob
+import json
+import logging
+import time
+from os import PathLike
+from pathlib import Path
+from typing import Callable, List, Optional, Union
+from uuid import uuid4
+
+from blackboxopt import (
+    Evaluation,
+    EvaluationSpecification,
+    Objective,
+    OptimizerNotReady,
+)
+from blackboxopt import logger as default_logger
+from blackboxopt.base import MultiObjectiveOptimizer, SingleObjectiveOptimizer
+from blackboxopt.optimization_loops.utils import (
+    evaluation_function_wrapper,
+    init_max_evaluations_with_limit_logging,
+)
+
+
+def run_optimization_loop(
+    optimizer: Union[SingleObjectiveOptimizer, MultiObjectiveOptimizer],
+    target_directory: PathLike,
+    timeout_s: float = float("inf"),
+    max_evaluations: Optional[int] = None,
+    proposal_queue_size: int = 1,
+    logger: Optional[logging.Logger] = None,
+) -> List[Evaluation]:
+    """
+    Args:
+        optimizer: The optimizer instance to use for specification generation and
+            evaluation ingestion.
+        target_directory: The directory where the evaluation specifications and results
+            are stored.
+        timeout_s: The maximum time in seconds to run the optimization loop.
+        max_evaluations: The maximum number of evaluations to perform.
+        proposal_queue_size: The number of proposals to keep in the queue, i.e. the size
+            of the evaluation batch for parallel/batch evaluation.
+        logger: The logger to use for logging. If `None`, the default logger is used.
+    """
+    if logger is None:
+        logger = default_logger
+
+    target_directory = Path(target_directory)
+    objectives = (
+        optimizer.objectives
+        if isinstance(optimizer, MultiObjectiveOptimizer)
+        else [optimizer.objective]
+    )
+    evaluations: List[Evaluation] = []
+
+    _max_evaluations = init_max_evaluations_with_limit_logging(
+        max_evaluations=max_evaluations, timeout_s=timeout_s, logger=logger
+    )
+
+    start = time.time()
+    while True:
+        evaluations_to_report = glob.glob(str(target_directory / "eval_result_*.json"))
+        for eval_result in evaluations_to_report:
+            with open(eval_result, "r", encoding="utf-8") as fh:
+                evaluation = Evaluation(**json.load(fh))
+            Path(eval_result).unlink(missing_ok=True)
```

Comment:
Why is it ok, if the file is missing? Is this simply for robustness, or is there another mechanism that should delete this file?